### DISCUSSION TITLE
I thought that `useElementVisibility` is using Intersection Observer API inside.

### DISCUSSION THREAD

**Discussion by ivanq3w** [vueuse/vueuse#651] [Q&A]
Just like in https://github.com/Akryum/vue-observe-visibility
And yes, `vue-observe-visibilty` works with vue3: https://github.com/Akryum/vue-observe-visibility/issues/219

But `useElementVisibility` works on window scroll event: https://github.com/vueuse/vueuse/blob/main/packages/core/useElementVisibility/index.ts

It is not clear from the docs : )

**Comment by wheatjs** [comment]
We have [`useIntersectionObserver`](https://vueuse.org/core/useIntersectionObserver/) if that is what you are looking for

**Reply by ivanq3w** [reply]
Oh, so many functions in vueuse, missed it : )

**Reply by curtgrimes** [reply]
@wheatjs Is there a reason why `useElementVisibility` internally uses a scroll event handler instead of an intersection observer? Would you be possibly open to PRs making `useElementVisibility` internally use `useIntersectionObserver`?

**Reply by curtgrimes** [reply]
I submitted PR https://github.com/vueuse/vueuse/pull/2551 for making `useElementVisibility()` internally use `useIntersectionObserver()` instead of a scroll event handler.
